src/main/java/io/getunleash/repository/HttpFeatureFetcher.java

```diff
@@ -59,30 +61,24 @@ private ClientFeaturesResponse getFeatureResponse(
                             new InputStreamReader(
                                     (InputStream) request.getContent(), StandardCharsets.UTF_8))) {
 
-                FeatureCollection features = JsonFeatureParser.fromJson(reader);
-                return new ClientFeaturesResponse(
-                        ClientFeaturesResponse.Status.CHANGED,
-                        features.getToggleCollection(),
-                        features.getSegmentCollection());
+                String clientFeatures = reader.lines().collect(Collectors.joining("\n"));
+
+                return ClientFeaturesResponse.updated(clientFeatures);
             }
         } else if (followRedirect
                 && (responseCode == HttpURLConnection.HTTP_MOVED_TEMP
                         || responseCode == HttpURLConnection.HTTP_MOVED_PERM
                         || responseCode == HttpURLConnection.HTTP_SEE_OTHER)) {
             return followRedirect(request);
         } else if (responseCode == HttpURLConnection.HTTP_NOT_MODIFIED) {
-            return new ClientFeaturesResponse(
-                    ClientFeaturesResponse.Status.NOT_CHANGED, responseCode);
+            return ClientFeaturesResponse.notChanged();
         } else {
-            return new ClientFeaturesResponse(
-                    ClientFeaturesResponse.Status.UNAVAILABLE,
-                    responseCode,
-                    getLocationHeader(request));
+            return ClientFeaturesResponse.unavailable(responseCode, getLocationHeader(request));
         }
     }
 
     private ClientFeaturesResponse followRedirect(HttpURLConnection request) throws IOException {
-        String newUrl = getLocationHeader(request);
+        String newUrl = getLocationHeader(request).get();
```

Comment:
If null this will fail hard, just details:
```suggestion
        String newUrl = getLocationHeader(request).orElseThrow(() -> new IllegalStateException("No Location header found in redirect response."));
```